web/packages/shared/libs/tdp/client.ts

```diff
@@ -706,12 +708,13 @@ export class TdpClient extends EventEmitter {
     this.send(msg);
   }
 
-  addSharedDirectory(sharedDirectory: FileSystemDirectoryHandle) {
-    this.sdManager.add(sharedDirectory);
+  async shareDirectory() {
+    await this.sharedDirectoryAccess.selectDirectory();
+    this.sendSharedDirectoryAnnounce();
```

Comment:
Why is this line needed now?

Reply:
It was previously called in `useDesktopSession` after selecting the directory 
https://github.com/gravitational/teleport/blob/b3de2bf9b4569375f8463c799ac476c844ca0be6/web/packages/shared/components/DesktopSession/useDesktopSession.tsx#L126-L135
I moved it here to make this API a more high level (so that selecting a directory and actually sharing it is a single thing).